### PR TITLE
Ports: Add libjodycode and jdupes

### PR DESCRIPTION
**Pull request by tarob0ba**
This ports the file deduplication utility `jdupes`.
There may be minor issues, but basic file deduplication tasks work well (and quick!).

### CODE REVIEW

File: Ports/AvailablePorts.md [hunk 1]
```diff
@@ -133,6 +133,7 @@ This list is also available at [ports.serenityos.net](https://ports.serenityos.n
 | [`libgpg-error`](libgpg-error/)                     | libgpg-error                                                    | 1.45                     | https://gnupg.org/software/libgpg-error/index.html                             |
 | [`libiconv`](libiconv/)                             | GNU libiconv                                                    | 1.17                     | https://www.gnu.org/software/libiconv/                                         |
 | [`libicu`](libicu/)                                 | ICU                                                             | 69.1                     | http://site.icu-project.org/                                                   |
+| [`libjodycode`](libjodycode/)                       | libjodycode                                                     | 3.1                      | hhttps://github.com/jbruchon/libjodycode                                       |
```
Comment (ADKaster) [inline Ports/AvailablePorts.md:137]: there's an hextra h here on the h-url

File: Ports/libjodycode/package.sh [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='libjodycode'
+version='3.1'
+files="https://github.com/jbruchon/libjodycode/archive/refs/tags/v${version}.tar.gz libjodycode-${version}.tar.gz c72974eb1d38873e06ea84b3d78990f87192f0113da5bd13fcac6bbc6a6e2184"
+auth_type='sha256'
+workdir="libjodycode-${version}"
```
Comment (ADKaster) [inline Ports/libjodycode/package.sh:6]: Given this: 
https://github.com/jbruchon/libjodycode/blob/master/Makefile#L26

this package will behave particularly badly if we try to cross-compile from aarch64-macOS to x86_64-serenity.

we should be able to override `UNAME_M`, `UNAME_S` and `CROSS_DETECT` in `makeopts()` though.

File: Ports/jdupes/package.sh [hunk 3]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env -S bash ../.port_include.sh
+port='jdupes'
+version='1.26.1'
+files="https://github.com/jbruchon/jdupes/archive/refs/tags/v${version}.tar.gz jdupes-${version}.tar.gz 09153824320c65ad529f8f97cd3b7e792c50e9f9018192ea1a76f2e33a196225"
+auth_type='sha256'
+workdir="jdupes-${version}"
+depends=('libjodycode')
+makeopts+=('DISABLE_DEDUPE=1')
```
Comment (ADKaster) [inline Ports/jdupes/package.sh:8]: this port has the same issue as the previous one, re `UNAME_S`

File: Ports/AvailablePorts.md (hunk 1 above)
Comment (tarob0ba) [reply Ports/AvailablePorts.md]: whoops!

File: Ports/libjodycode/package.sh (hunk 2 above)
Comment (tarob0ba) [reply Ports/libjodycode/package.sh:6]: Alright, I've overridden these values and everything compiles like before. Should I make any more modifications?

File: Ports/jdupes/package.sh (hunk 3 above)
Comment (tarob0ba) [reply Ports/jdupes/package.sh]: Addressed. Does this need any further changes?

**Review by ADKaster — APPROVED** [review]
Tried this out locally and it seems to work, nice job 👍

**Pull request merged**